nix/build-haskell-package.nix

```diff
@@ -31,6 +45,19 @@ lib.pipe root
     (mkNewStorePath "source-${name}")
     (x: log.traceDebug "${name}.mkNewStorePath ${x.outPath}" x)
 
-    (root: self.callCabal2nix name root { })
-    (x: log.traceDebug "${name}.cabal2nixDeriver ${x.cabal2nixDeriver.outPath}" x)
+     (root:
+        # Check if cached cabal2nix generated nix expression is present,
+        # if present use it with callPackage
+        # to avoid IFD
+        let pkg = 
+          if lib.pathExists (lib.concatStringsSep "/" [ root "default.nix" ])
```

Comment:
The name of this file should be configurable by the user, much like rust-flake's [`crateNixFile`](https://github.com/juspay/rust-flake/blob/cdcm17a1d009674a9a23c7e4749baf474d2419f04/nix/modules/flake-module.nix#L54).

Also, the default value of this option should not be `default.nix` (people use it for a lot of other things), but rather something ore unique, like `cabal.nix`.

Reply:
You can open a PR on https://github.com/cachix/git-hooks.nix adding an option that allows the user to override it.

Reply:
https://github.com/cachix/git-hooks.nix/pull/531